### PR TITLE
Exporter/Metrics/OcAgent: Add worker thread and exporter.

### PR DESCRIPTION
**Pull request by songy23**
~~Blocked by https://github.com/census-instrumentation/opencensus-java/pull/1637.~~

Fixes https://github.com/census-instrumentation/opencensus-java/issues/1567.

### CODE REVIEW

**Comment by bogdandrutu** [comment]
Please rebase if this PR still needs to get merged.

**Comment by songy23** [comment]
Sure, this PR is blocked by https://github.com/census-instrumentation/opencensus-java/pull/1637. Will rebase once we have an agreement there.

**Comment by bogdandrutu** [comment]
Can we unblock this PR without submitting the other one?

**Comment by songy23** [comment]
> Can we unblock this PR without submitting the other one?

Yes, I removed the caching for now. PTAL.

File: exporters/metrics/ocagent/src/main/java/io/opencensus/exporter/metrics/ocagent/OcAgentMetricsExporterWorker.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+  private void connect() {
+    ManagedChannelBuilder<?> channelBuilder = ManagedChannelBuilder.forTarget(endPoint);
+    if (useInsecure) {
+      channelBuilder.usePlaintext();
+    }
+    ManagedChannel channel = channelBuilder.build();
+    MetricsServiceGrpc.MetricsServiceStub stub = MetricsServiceGrpc.newStub(channel);
+    exportRpcHandler = OcAgentMetricsServiceExportRpcHandler.create(stub);
+    ExportMetricsServiceRequest firstRequest =
```
Comment (rghetia) [inline exporters/metrics/ocagent/src/main/java/io/opencensus/exporter/metrics/ocagent/OcAgentMetricsExporterWorker.java]: what is the purpose of sending firstRequest without any metrics?

File: exporters/metrics/ocagent/src/main/java/io/opencensus/exporter/metrics/ocagent/OcAgentMetricsExporter.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * The implementation of the OpenCensus Agent (OC-Agent) Metrics Exporter.
+ *
+ * <p>Example of usage:
+ *
+ * <pre>{@code
+ * public static void main(String[] args) {
+ *   OcAgentMetricsExporter.createAndRegister();
```
[the remark this answers is not in the document]
Comment (songy23) [reply exporters/metrics/ocagent/src/main/java/io/opencensus/exporter/metrics/ocagent/OcAgentMetricsExporter.java]: Yes, updated. Good catch.

File: exporters/metrics/ocagent/src/main/java/io/opencensus/exporter/metrics/ocagent/OcAgentMetricsExporter.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      @Nullable String serviceName,
+      @Nullable Duration exportInterval,
+      @Nullable Duration retryInterval) {
+    if (endPoint == null) {
+      endPoint = DEFAULT_END_POINT;
+    }
+    if (useInsecure == null) {
+      useInsecure = false;
+    }
+    if (serviceName == null) {
```
[the remark this answers is not in the document]
Comment (songy23) [reply exporters/metrics/ocagent/src/main/java/io/opencensus/exporter/metrics/ocagent/OcAgentMetricsExporter.java]: It shouldn't be empty. Added a check here.

File: exporters/metrics/ocagent/src/main/java/io/opencensus/exporter/metrics/ocagent/OcAgentMetricsExporterWorker.java (hunk 1 above)
Comment (songy23) [reply exporters/metrics/ocagent/src/main/java/io/opencensus/exporter/metrics/ocagent/OcAgentMetricsExporterWorker.java]: The first message is used to open the RPC and provide initial metadata to Agent. E.g `Node` is needed only in the first message. After that RPC will be kept open and `Node` can be omitted in the subsequent messages.

Comment (rghetia) [reply exporters/metrics/ocagent/src/main/java/io/opencensus/exporter/metrics/ocagent/OcAgentMetricsExporterWorker.java]: SG.

Comment (rghetia) [reply exporters/metrics/ocagent/src/main/java/io/opencensus/exporter/metrics/ocagent/OcAgentMetricsExporterWorker.java]: I don't have privilege to approve it.

**Comment by rghetia** [comment]
I don't have privilege to approve it.

**Comment by bogdandrutu** [comment]
@rghetia what do you mean? I think you have write access.

**Comment by rghetia** [comment]
> @rghetia what do you mean? I think you have write access.

Not sure how I didn't see the approve option ;-). I can approve.

**Comment by songy23** [comment]
Filed https://github.com/census-instrumentation/opencensus-java/issues/1657 to update this to use `metrics-util`. Merging this PR first. Thanks for reviewing!

**Pull request merged**